src/v3/plugins/github/address.test.js

```diff
@@ -0,0 +1,209 @@
+// @flow
+
+import * as GithubAddress from "./address";
+import {NodeAddress} from "../../core/graph";
+
+describe("plugins/github/address", () => {
+  const structure = GithubAddress.structure;
+  const destructure = GithubAddress.destructure;
+  const repo = () => ({
+    type: "REPO",
+    owner: "sourcecred",
+    name: "example-github",
+  });
+  const issue = () => ({type: "ISSUE", repo: repo(), number: "2"});
+  const pull = () => ({type: "PULL", repo: repo(), number: "5"});
+  const review = () => ({type: "REVIEW", pull: pull(), fragment: "100313899"});
+  const issueComment = () => ({
+    type: "COMMENT",
+    parent: issue(),
+    fragment: "373768703",
+  });
+  const pullComment = () => ({
+    type: "COMMENT",
+    parent: pull(),
+    fragment: "396430464",
+  });
+  const reviewComment = () => ({
+    type: "COMMENT",
+    parent: review(),
+    fragment: "171460198",
+  });
+  const user = () => ({type: "USERLIKE", login: "decentralion"});
+  const examples = {
+    repo,
+    issue,
+    pull,
+    review,
+    issueComment,
+    pullComment,
+    reviewComment,
+    user,
+  };
+
+  describe("Structured -> Raw -> Structured is identity", () => {
+    Object.keys(examples).forEach((example) => {
+      it(example, () => {
+        const instance = examples[example]();
+        expect(structure(destructure(instance))).toEqual(instance);
+      });
+    });
+  });
+
+  describe("Raw -> Structured -> Raw is identity", () => {
+    Object.keys(examples).forEach((example) => {
+      it(example, () => {
+        const instance = examples[example]();
+        const raw = destructure(instance);
+        expect(destructure(structure(raw))).toEqual(raw);
+      });
+    });
+  });
+
+  describe("snapshots as expected:", () => {
+    Object.keys(examples).forEach((example) => {
+      it(example, () => {
+        const instance = examples[example]();
+        const raw = NodeAddress.toParts(destructure(instance));
+        expect({address: raw, structured: instance}).toMatchSnapshot();
+      });
+    });
+  });
+
+  describe("errors on", () => {
+    describe("structure() with", () => {
```

Comment:
nit: Can we write `structure(...)` or `` `structure` `` instead of `structure()`?

Reply:
Done.